### PR TITLE
Feature/identity sign in with user information

### PR DESCRIPTION
**Pull request by jaelim-ms**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Adding a completionHandler as a parameter to signIn method.

### CODE REVIEW

File: AppCenterIdentity/AppCenterIdentity/MSIdentity.h [hunk 1]
```diff
@@ -1,7 +1,35 @@
...
+ */
+typedef void (^MSSignInCompletionHandler)(MSUserInformation *_Nullable userInformation, NSError *_Nullable error);
+
+/**
+ * Error code for Identity.
+ */
+typedef NS_ENUM(NSInteger, MSIdentityErrorCode) {
+  MSIdentityErrorServiceDisabled = -420000,
+
+  MSIdentityErrorPreviousSignInRequestNotCompleted = -420001
```
Comment (dhei) [inline AppCenterIdentity/AppCenterIdentity/MSIdentity.h]: maybe `MSIdentityErrorAnotherSignInInProgress` is more clear about the purpose? 😄

Comment (guperrot) [reply AppCenterIdentity/AppCenterIdentity/MSIdentity.h]: Are you sure? The double In does not look better.

Comment (ElektrojungeMS) [reply AppCenterIdentity/AppCenterIdentity/MSIdentity.h]: I agree that Di's proposal is clearer about the intention.
How about `MSIdentityErrorAnotherSignInIsInProgress`? It's already long and `InIsIn` reads a little less weird than `InIn`?
My favorite would be `MSIdentityErrorPreviousSignInRequestInProgress`?

Comment (guperrot) [reply AppCenterIdentity/AppCenterIdentity/MSIdentity.h]: I prefer the latter

File: AppCenterCrashes/AppCenterCrashes/MSCrashes.mm [hunk 2]
```diff
@@ -174,7 +174,7 @@ + (void)generateTestCrash {
   @synchronized([MSCrashes sharedInstance]) {
     if ([[MSCrashes sharedInstance] canBeUsed]) {
       if ([MSUtility currentAppEnvironment] == MSEnvironmentAppStore) {
-          MSLogWarning([MSCrashes logTag], @"GenerateTestCrash was just called in an App Store environment. The call will be ignored");
+        MSLogWarning([MSCrashes logTag], @"GenerateTestCrash was just called in an App Store environment. The call will be ignored");
```
Comment (ElektrojungeMS) [inline AppCenterCrashes/AppCenterCrashes/MSCrashes.mm]: We're missing a `.` inside the comment. Not technically part of this PR but let's fix it now.

File: AppCenterIdentity/AppCenterIdentity/MSIdentity.m [hunk 3]
```diff
@@ -147,10 +151,30 @@ + (BOOL)openURL:(NSURL *)url {
   return [MSALPublicClientApplication handleMSALResponse:url];
 }
 
-+ (void)signIn {
++ (void)signInWithCompletionHandler:(MSSignInCompletionHandler _Nullable)completionHandler {
+
+  // We allow completion handler to be optional but we need a non nil one to track operation progress internally.
```
Comment (ElektrojungeMS) [inline AppCenterIdentity/AppCenterIdentity/MSIdentity.m]: Thx for documenting this. It's quite counter-intuitive.

**Pull request merged**